### DISCUSSION TITLE
Example showing how to process a web_sys::File in web workers

### DISCUSSION THREAD

**Discussion by LucaCappelletti94** [rustwasm/gloo#480] [Q&A]
All web workers made available in [gloo worker](https://docs.rs/gloo-worker/latest/gloo_worker/) require for the arguments to be serializable and [`web_sys::File`](https://docs.rs/web-sys/latest/web_sys/struct.File.html) does not implement either [`serde::Serialize`](https://docs.rs/serde/latest/serde/trait.Serialize.html) or [`serde::Deserialize`](https://docs.rs/serde/latest/serde/trait.Deserialize.html).

It seems to me that files can only be processed in the web worker after having been fully loaded and the bincode serialized and submitted, which may block the page execution. This should not be the case as, apparently, in javascript there is a [`Worker.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) that can accept a File object, and this method is available in the web-sys bindings here: [`post_message`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Worker.html#method.post_message). So, how do I process a File with a web worker?

**Comment by sorz** [comment]
I do not use gloo but if you are looking for a way to serialize web_sys::File…

serde-wasm-bindgen dose provide a way to serde any JS value. Put `#[serde(with = "serde_wasm_bindgen::preserve")]` over `File` field should do the trick. See the [Preserving JavaScript values](https://docs.rs/serde-wasm-bindgen/latest/serde_wasm_bindgen/#preserving-javascript-values) section.

**Reply by LucaCappelletti94** [reply]
I will check it out and update this discussion accordingly!
